Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,239 @@
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void AddAttribute_NullPlainTextWebString_RendersEmptyAttribute ()
+    {
+      var webString = WebString.CreateFromText (null);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"\">"));
+    }
+
+    [Test]
+    public void AddAttribute_EmptyPlainTextWebString_RendersEmptyAttribute ()
+    {
+      var webString = WebString.CreateFromText (string.Empty);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"\">"));
+    }
+
+    [Test]
+    public void AddAttribute_PlainTextWebStringWithAttributeEnum_RendersEncodedAttributeValue ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromText (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"aoe   &quot; &amp; &#39; &lt; > é \r \n \r\n\">"));
+    }
+
+    [Test]
+    public void AddAttribute_PlainTextWebStringWithAttributeEnumThatDoesNotHaveEncodeDefault_RendersEncodedAttributeValue ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromText (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Cols);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a cols=\"aoe   &quot; &amp; &#39; &lt; > é \r \n \r\n\">"));
+    }
+
+    [Test]
+    public void AddAttribute_PlainTextWebStringWithAttributeString_RendersEncodedAttributeValue ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromText (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, "class");
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"aoe   &quot; &amp; &#39; &lt; > é \r \n \r\n\">"));
+    }
+
+    [Test]
+    public void AddAttribute_EncodedWebStringWithAttributeString_RendersUnencodedAttributeValue ()
```

Comment:
Are we mssing the test with CreateFromHtml for Attribute.Class and Attribute.Cols ?